libfreerdp/core/multitransport.c

```diff
@@ -63,13 +63,20 @@ state_run_t multitransport_recv_request(rdpMultitransport* multi, wStream* s)
 
 	UINT32 requestId;
 	UINT16 requestedProto;
+	UINT16 reserved;
 	const BYTE* cookie;
 
 	Stream_Read_UINT32(s, requestId);      /* requestId (4 bytes) */
 	Stream_Read_UINT16(s, requestedProto); /* requestedProtocol (2 bytes) */
-	Stream_Seek(s, 2);                     /* reserved (2 bytes) */
+	Stream_Read_UINT16(s, reserved);       /* reserved (2 bytes) */
 	cookie = Stream_ConstPointer(s);
 	Stream_Seek(s, RDPUDP_COOKIE_LEN); /* securityCookie (16 bytes) */
+	if (reserved != 0)
+	{
+		WLog_WARN(TAG, "reserved is %d instead of 0, skipping %d bytes of unknown data", reserved,
+		          Stream_GetRemainingLength(s));
+		Stream_Seek(s, Stream_GetRemainingLength(s));
```

Comment:
`Stream_SafeSeek` might be appropriate, as we don´t know if the stream actually contains all that data

Reply:
I disagree except if I miss something `Stream_Seek(s, Stream_GetRemainingLength(s));` is always safe as Safe_Seek just checks `Stream_GetRemainingLength(s)`

Reply:
@hardening true, forgot to add that with the log above `reserved` should contain the amount of remaining data?

Reply:
reserved does actually not contain the length but 1. So probably reserved is now a version field.